app/controllers/api/results_controller.rb

```diff
@@ -6,6 +6,77 @@ def results_search_conditions
       MiqReportResult.for_user(User.current_user).where_clause.ast
     end
 
+    def apply_limit_and_offset(results, options)
+      results.slice(options['offset'].to_i, options['limit'].to_i) || []
+    end
+
+    def sort_params_from(report_result)
+      params['sort_order'] ||= :ascending
+      params['sort_order'] = :descending if params['sort_order'] == 'desc'
+
+      sort_colums = sort_by(report_result).split(",").collect do |attr|
+        if report(report_result).col_order&.include?(attr)
+          attr
+        else
+          raise BadRequestError, "#{attr} is not a valid attribute for #{report_result.name}"
+        end
+      end.compact
+
+      {:sort_by => sort_colums, :sort_order => (params['sort_order'] || :ascending).to_sym }
```

Comment:
up to you, but I was thinking you could just return the values themselves

It would be easier for me to read on the other end

```ruby
sort_by, sort_order = sort_params_from(report_result)
```

```suggestion
      [sort_colums, (params['sort_order'] || :ascending).to_sym]
```